lib/parsers/regex.js

```diff
@@ -0,0 +1,55 @@
+'use strict';
+const inherits = require('util').inherits;
+const Transform = require('stream').Transform;
+
+function RegexParser(options) {
+  if (!(this instanceof RegexParser)) {
+    return new RegexParser(options);
+  }
+  Transform.call(this, options);
+
+  options = options || {};
+
+  if (options.delimiter === undefined) {
+    throw new TypeError('"delimiter" is not a RegExp object or a string');
+  }
+
+  if (options.delimiter.length === 0) {
+    throw new TypeError('"delimiter" has a 0 or undefined length');
+  }
+
+  if (!(options.delimiter instanceof RegExp)) {
+    options.delimiter = new RegExp(options.delimiter);
+  }
+
+  const encoding = options.encoding || 'utf8';
+  this.setEncoding(encoding);
+
+  this.delimiter = options.delimiter;
+  this.buffer = new Buffer(0);
+}
+
+inherits(RegexParser, Transform);
+
+RegexParser.prototype._transform = function(chunk, encoding, cb) {
+  let data = Buffer.concat([this.buffer, chunk]).toString();
+
+  const lines = data.split(this.delimiter);
```

Comment:
Makes sense! Reverted to `parts` which was used in the old node-serialport version.